onnxscript/version_converter/version_converter.py

```diff
@@ -1,55 +1,427 @@
+        replacement = self.process_node(node, opset_version)
+        if replacement is None:
+            # No change. Process attributes.
+            for attr in node.attributes.values():
+                self.visit_attribute(attr, opset_version)
+            return None
+        else:
+            self.replace_node(node, replacement, root)
+
+    def visit_graph(self, graph: ir.Graph, opset_version: int) -> None:
         for node in graph:
-            node.version = self.model_version
+            self.visit_node(node, graph, opset_version)
+            node.version = self.target_version
+
+    def visit_function(self, function: ir.Function, opset_version: int) -> None:
+        for node in function:
+            self.visit_node(node, function, opset_version)
+
+    def visit_model(self, model: ir.Model) -> None:
+        self.opset_imports = model.opset_imports
+        model_version = model.opset_imports.get("")
 
+        if self.target_version < model_version:
+            logger.warning(
+                "Target opset: %s less than %s, downstream version conversion not currently supported.",
+                self.target_version,
+                model_version,
+            )
+            return
         # Iterate from current model version -> target version
         # Updating each node based on the correct adapter [ver->ver+1]
-        for opset_version in range(self.model_version, target_version):
-            for node in graph:
-                adapter_set = pick_adapter_set(opset_version)
-                if node.op_type in adapter_set:
-                    adapter_func = adapter_set[node.op_type]
-                    if adapter_func is not None:
-                        adapter_func(node)
-                    node.version = opset_version + 1
-                # Apply custom adapters
-                if (
-                    self.custom_adapters is not None
-                    and str(opset_version) in self.custom_adapters
-                ):
-                    custom_adapter_set = self.custom_adapters[str(opset_version)]
-                    if node.op_type in custom_adapter_set:
-                        adapter_func = custom_adapter_set[node.op_type]
-                        if adapter_func is not None:
-                            adapter_func(node)
-
-
-def version_convert(model: ir.Model, target_version: int, custom_adapters=None) -> None:
+        for opset_version in range(model_version, self.target_version):
+            if tuple((opset_version, opset_version + 1)) not in _ADAPTER_SETS:
+                logger.warning(
+                    "Conversion from opset: %s to target opset: %s not currently supported.",
+                    opset_version,
+                    opset_version + 1,
+                )
+                return
+
+            self.visit_graph(model.graph, opset_version)
+            for function in model.functions.values():
```

Comment:
As explained in earlier comment, I think it is better to skip functions (they won't be handled correctly). Instead, we should call the inliner to inline all functions as the first step of version_conversion (probably in the caller, before line 426 below).

Reply:
> they won't be handled correctly

Curious why that would be the case?